src/wp-admin/includes/class-wp-upgrader.php

```diff
@@ -798,6 +798,15 @@ public function run( $options ) {
 			)
 		);
 
+		/**
+		 * Filter result of install_package().
+		 *
+		 * @since 5.7.0
+		 * @param bool|WP_Error $result  Result from install_package().
+		 * @param array         $options Array of data for plugin/theme being updated.
+		 */
+		$result = apply_filters( 'upgrader_install_complete', $result, $options );
```

Comment:
Maybe `upgrader_install_package_result` would be more appropriate for it's context? Longer name for sure but it's easier to identify.

Reply:
Naming these things is usually the hardest part.